### PR TITLE
Add support for Terraform v0.15

### PR DESCRIPTION
**Pull request by soerenmartius**
### Added
- feat: add support for terraform `v0.15.x`
### Changed
- refactor: refactor build-tools to use named volumes for go and terraform files
- build: upgrade build-tools to `v0.11.0`
- build: update secrets in github actions pipeline
- feat: updates aws provider version in minimal test scenario
- feat: upgrade terratest to v0.34.0
- build: upgrade pre-commit-hooks to v0.2.2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ Most basic usage just setting required arguments:
 ```hcl
 module "terraform-aws-vpc" {
   source  = "mineiros-io/vpc/aws"
-  version = "~> 0.2.0"
+  version = "~> 0.3.1"
```
Comment (mariux) [inline README.md:77]: whats's the next version should be 0.3.0 ?

**Review by mariux — CHANGES_REQUESTED** [review]

File: versions.tf [hunk 2]
```diff
@@ -1,5 +1,9 @@
+# ------------------------------------------------------------------------------
+# SET TERRAFORM AND PROVIDER REQUIREMENTS FOR RUNNING THIS MODULE
+# ------------------------------------------------------------------------------
+
 terraform {
-  required_version = ">= 0.12.20, < 0.14"
+  required_version = ">= 0.12.20, < 0.16"
```
Comment (mariux) [inline versions.tf:6]: missing the test against 0.14 in the commit history

File: variables.tf [hunk 3]
```diff
@@ -161,7 +161,7 @@ variable "module_enabled" {
 }
 
 variable "module_depends_on" {
-  type        = list
+  type        = list(any)
```
Comment (mariux) [inline variables.tf:164]:
```suggestion
  type        = any
```

**Review by mariux — CHANGES_REQUESTED** [review]

File: versions.tf (hunk 2 above)
Comment (soerenmartius) [reply versions.tf:6]: should be backward compatible? We tested against v0.14 in the other PR which was closed in favor of this one

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Changed
+
+- refactor: refactor build-tools to use named volumes for go and terraform files
+- build: upgrade build-tools to `v0.11.0`
+- build: update secrets in github actions pipeline
+- feat: updates aws provider version in minimal test scenario
+- feat: upgrade terratest to v0.34.0
+- build: upgrade pre-commit-hooks to v0.2.2
+
```
Comment (mariux) [inline CHANGELOG.md:32]:
```suggestion
### Added

- Add support for Terraform `v0.15`

```

**Review by mariux — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,6 +17,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - BREAKING: Fix terraform route modification behavior when switching nat_gateway_mode
 
+### Added
+
+- Add support for Terraform `v0.15`
+
```
Comment (mariux) [inline CHANGELOG.md:23]: added should go before fixed... we shoudl release 0.14. as a specific version or at least reserve a version number for it before 0.15

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**